connect/basic-auth-extension/src/main/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilter.java

```diff
@@ -174,4 +153,84 @@ public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
                 ));
         }
     }
+
+    private void setSecurityContextForRequest(ContainerRequestContext requestContext, BasicAuthenticationCredential credential) {
+        requestContext.setSecurityContext(new SecurityContext() {
+            @Override
+            public Principal getUserPrincipal() {
```

Comment:
Note: The Javadoc indicates that this method should only return a non-null Principal if authentication has already been set. This invariant is preserved when `setSecurityContextForRequest` is called explicitly only after the `LoginContext::login` method successfully exits, which it is.

:+1: